ASIM/dev/ASimTester/filteringTest/filteringTest.py

```diff
@@ -0,0 +1,227 @@
+
+
+def create_call_with_parameter(parameter, value, column_name):
+    return f"query({parameter}={value}) | summarize count() by {column_name}\n"
+
+
+class FilteringTest(unittest.TestCase):
+    # "Main" function which runs initiates the validations 
+    def tests_main_func(self):
+        if not os.path.exists(parser_file_path):
+            self.fail(f"File path does not exist: {parser_file_path}")
+        if not parser_file_path.endswith('.yaml'): 
+            self.fail(f"Not a yaml file: {parser_file_path}")
+        try:
+            parser_file = get_parser(parser_file_path)
+        except:
+            self.fail(f"Cannot open file: {parser_file_path}")
+        self.check_required_fields(parser_file)
+        no_call_query = create_query_without_call(parser_file)
+        self.check_data_in_workspace(no_call_query)
+        columns_in_answer = self.get_columns_of_parser_answer(no_call_query)
+        if parser_file['Normalization']['Schema'] not in all_schemas_parameters:
+            self.fail(f"Schema: {parser_file['Normalization']['Schema']} - is not supported")
+        param_to_column_mapping = all_schemas_parameters[parser_file['Normalization']['Schema']]
+        for param in parser_file['ParserParams']:
+            with self.subTest():
+                if param['Name'] not in param_to_column_mapping:
+                    self.fail(f"parameter: {param['Name']} - is not a valid parameter")
+                column_name_in_table = param_to_column_mapping[param['Name']]
+                self.handle_param(param, no_call_query, columns_in_answer, column_name_in_table)            
+                      
+    # Sending a parameters to the suitable test according to the parameter type    
+    def handle_param(self, param, no_call_query, columns_in_answer, column_name_in_table):
+        param_name = param['Name']
+        param_type = param['Type']
+        if (param_name == "pack"):
+            return 
+        if (param_name == "disabled"):
+            self.disabled_test(param, no_call_query)
+        elif  column_name_in_table not in columns_in_answer:
+            return
+        elif (param_type == "datetime"):
+            pass
+        elif (param_type == "dynamic"):
+            pass
+        else:
+            self.scalar_test(param, no_call_query, column_name_in_table)
+
+    
+    # Checking if the provided workspace has some data
+    def check_data_in_workspace(self, no_call_query):
+        check_data_response = self.send_query(no_call_query + "query() | take 5")
+        if len(check_data_response.tables[0].rows) == 0:
+            self.fail("No data in the provided workspace")
+          
+
+    # Checking if all fields from "required_fields" array are in the yaml file.
+    def check_required_fields(self, parser_file):
+        missing_fields = []
+        for full_field in required_fields:
+            file = parser_file
+            fields = full_field.split('.')
+            for field_name in fields:
```

Comment:
Why do you need this? Why can't you define it split in advance?

Reply:
But you don't get the required_fields as input, you set it in advance, why can't you just set it to be a list of fields?